projects/vdk-plugins/vdk-kerberos-auth/src/vdk/plugin/kerberos/kerberos_configuration.py

```diff
@@ -101,3 +111,12 @@ def add_definitions(config_builder: ConfigurationBuilder) -> None:
         description="Specifies the name of the Kerberos KDC (Key Distribution Center) host. "
         "This value is used only with the 'minikerberos' authentication type.",
     )
+    config_builder.add(
+        key=KRB_AUTH_FAIL_FAST,
+        default_value=False,
+        description="Specify if the authenticator must raise exception "
+        "if it fails to successfully authenticate with kerberos (basically - kinit). "
+        "If set to false, only warning will be logged on authentication failure. "
+        "Subsequent kerberos related requests may fail but that'd fail lazily (on demand) "
+        "that makes it possible for non-kerberos related features to work.",
+    )
```

Comment:
I don't understand the need for this env flag. If not all commands require Kerberos auth, and thus would fail, in what use case would a user want the app to fail when auth fails?

Reply:
When trying to debug issues - making it possible to fail fast would make it easier to debug since they'd see the original error quickly. Or in some cloud runs where we know that job should not continue without authentication successful.